compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BComplexVarSymbol.java

```diff
@@ -0,0 +1,40 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.wso2.ballerinalang.compiler.semantics.model.symbols;
+
+import io.ballerina.tools.diagnostics.Location;
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.model.symbols.SymbolOrigin;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BType;
+import org.wso2.ballerinalang.compiler.util.Name;
+
+
+/**
+ * This symbol is used as the symbol of variables other than a simple variable.
+ * <br/>
+ * The only need of a symbol for complex variables is to record dependencies in the DataFlowAnalyzer.
```

Comment:
Can't we make this work with `` instead of introducing a new kind of symbol just for this?

Reply:
We can, but as discussed with @rdhananjaya it would be better to distinguish these symbols with simple variable symbols. because these symbols are only used in DataFlowAnalyzer.

Reply:
Used the same `BVarSymbol` for complex variables as well.